src/components/Badge/Badge.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { forwardRef, useRef } from "react";
+import cx from "classnames";
+import { useMergeRefs } from "../../hooks";
+import VibeComponentProps from "../../types/VibeComponentProps";
+import VibeComponent from "../../types/VibeComponent";
+import { getTestId } from "../../tests/test-ids-utils";
+import { ComponentDefaultTestId } from "../../tests/constants";
+import styles from "./Badge.module.scss";
+import { BadgeColor, BadgePosition, BadgeType } from "./BadgeConstants";
+import Indicator, { IndicatorProps } from "./Indicator/Indicator";
+import Counter, { CounterProps } from "../Counter/Counter";
+import { getStyle } from "../../helpers/typesciptCssModulesHelper";
+import { camelCase } from "lodash-es";
+import { IndicatorColor } from "./Indicator/IndicatorConstants";
+import { CounterColor } from "../Counter/CounterConstants";
+
+const BadgeColorToIndicatorColor = {
+  [BadgeColor.PRIMARY]: IndicatorColor.PRIMARY,
+  [BadgeColor.NOTIFICATION]: IndicatorColor.NOTIFICATION
+};
+
+const BadgeColorToCounterColor = {
+  [BadgeColor.PRIMARY]: CounterColor.PRIMARY,
+  [BadgeColor.NOTIFICATION]: CounterColor.NEGATIVE
+};
+
+export interface BadgeProps extends VibeComponentProps {
+  type?: BadgeType;
+  position?: BadgePosition;
+  color?: BadgeColor;
+  outbound?: boolean;
+  circular?: boolean;
+  border?: boolean;
+  badgeProps?: Omit<CounterProps, "color"> | Omit<IndicatorProps, "color">;
```

Comment:
shouldn't it be `counterProps`?

Reply:
it should receive any of the `type` option props. 
For now, it's either Counter or Indicator, I guess next we might have more (icons?).
So it should allow to inject props for any badge, to the component itself

Reply:
as a consumer of this component, I'd expect on of the following.
1. Use the `Badge` component and pass it direct props like `border`. so if one of its capabilities is displaying a counter, then I'd expect a prop called `counter` to be passed to it.
2. Use the `Badge` component and understand that behind the scenes it uses the `Counter` component, and thus I shall pass the `counterProps` so it will propagate.
3. Use the `Badge` component and compose `Counter` on my end, and then I know exactly which props to pass into the counter component.

The current state is a bit of a mish-mash, so I'd suggest to go either with option 1 or option 2

Reply:
@shlomitc Thanks to your suggestions I was able to achieve pretty solid type checking!
Please check the last commit and tell me what you think.

If user does not pass type of Counter, TS will allow only Indicator props (flat into regular props, not another object), if the user does pass type Counter, the user is able to only type in Counter props (again, flat).
Examples:

<img width="596" alt="image" src="https://github.com/mondaycom/monday-ui-react-core/assets/30905362/bc70ae35-1abe-4e0c-b6c2-448f28704a62">
<img width="936" alt="image" src="https://github.com/mondaycom/monday-ui-react-core/assets/30905362/a8ab815a-e1cc-489f-a4d3-db57521d31ea">
<img width="886" alt="image" src="https://github.com/mondaycom/monday-ui-react-core/assets/30905362/3c92f90a-7164-4af5-bb1c-18b2e9e059e5">

Reply:
of course you can also pass other Counter/Indicator props inline (flat)
e.g.:
```
<Badge type={Badge.types.COUNTER} count={100} maxDigits={2}>
  <Button leftIcon={WhatsNew}>What's new</Button>
</Badge>
```

Reply:
My personal preference is that components will be as dumb as possible, so this flattening is great already.